pkg/cache/tas_flavor_snapshot.go

```diff
@@ -887,51 +1004,113 @@ func useLeastFreeCapacityAlgorithm(unconstrained bool) bool {
 		(unconstrained && features.Enabled(features.TASProfileMixed))
 }
 
-func (s *TASFlavorSnapshot) updateSliceCountsToMinimum(domains []*domain, count int32, sliceSize int32, unconstrained bool) []*domain {
+func (s *TASFlavorSnapshot) updateSliceCountsToMinimum(domains []*domain, count int32, leaderCount int32, sliceSize int32, unconstrained bool) []*domain {
 	result := make([]*domain, 0)
 	remainingSlices := count / sliceSize
+	remainingLeaderCount := leaderCount
 	for i, domain := range domains {
-		if useBestFitAlgorithm(unconstrained) && domain.sliceState >= remainingSlices {
-			// optimize the last domain
-			domain = findBestFitDomainForSlices(domains[i:], remainingSlices)
-		}
+		if remainingLeaderCount > 0 {
```

Comment:
Let's try to generalize this if. Maybe we can get around it and have generic code.

Reply:
This request has been reflected here: https://github.com/kubernetes-sigs/kueue/issues/6244